python/tank/util/shotgun.py

```diff
@@ -382,12 +344,45 @@ def create_sg_connection(user="default"):
             "creating a shotgun API instance based on script based credentials in the "
             "shotgun.yml configuration file."
         )
-        config_data = __get_sg_config_data_with_script_user(__get_sg_config(), user)
-        api_handle = __create_sg_connection(config_data)
+
+        # try to find the shotgun.yml path
+        try:
+            config_file_path = __get_sg_config()
+        except TankError, e:
+            log.error(
+                "Trying to create a shotgun connection but this tk session does not have "
+                "an associated authenticated user. Therefore attempted to fall back on "
+                "a legacy authentication method where script based credentials are "
```

Comment:
yep, it's legacy - we want everyone to connect with user based auth because running toolkit with a single script user is a legacy setup. That doesn't mean that it's going away anytime soon - but we should think of it as a workflow that is a fallback that should be used only when user based doesn't work for whatever reason.